code/src/components/MessageBoard.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react'
+import moment from 'moment'
+
+import { API_URL, HEARTS_URL } from '../reusable/urls'
+
+import Loading from '../components/Loading'
+import ErrorBoard from '../components/ErrorBoard'
+import MyLikes from './MyLikes'
+
+const MessageBoard = ({ thoughts, setThoughts, myLikes, setMyLikes }) => {
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(false)
+
+  const updateBoard = (url) => {
+    setLoading(true)
+    fetch(url)
+      .then(response => response.json())
+      .then(receivedThoughts => {
+        setThoughts(receivedThoughts)
+        setTimeout(() => {
+          setLoading(false)
+        }, 2000)
+      })
+      .catch(err => {
+        setLoading(false)
+        setError(true)
+      })
+  }
+
+  useEffect(() => {
+    const fetchThoughts = (url) => {
+      setLoading(true)
+      fetch(url)
+        .then(response => response.json())
+        .then(receivedThoughts => {
+          setThoughts(receivedThoughts)
+          setTimeout(() => {
+            setLoading(false)
+          }, 2000)
+        })
+        .catch(err => {
+          setLoading(false)
+          setError(true)
+        })
+    }
```

Comment:
isn't it the same to call just `updateBoard()` here??

Reply:
So, yes, That was the question. I wanted to update board every time it is mounted. But then I noticed that unless the user refreshes the page, the new messages from other users do not pop-up, so I wanted to create a button that provides this option. Of course it is possible to just refresh the page, but it I wanted to have a button for it. Maybe silly idea ...